main.go

```diff
@@ -17,33 +20,96 @@ import (
 	"github.com/gocardless/draupnir/store"
 	"github.com/gocardless/draupnir/version"
 	"github.com/gorilla/mux"
-	"github.com/kelseyhightower/envconfig"
 	"github.com/oklog/run"
+	"github.com/pkg/errors"
 )
 
 type Config struct {
-	Port                   int    `required:"true"`
-	InsecurePort           int    `required:"false" default:"8080"`
-	DatabaseUrl            string `required:"true" split_words:"true"`
-	DataPath               string `required:"true" split_words:"true"`
-	Environment            string `required:"false"`
-	SharedSecret           string `required:"true" split_words:"true"`
-	OauthRedirectUrl       string `required:"true" split_words:"true"`
-	OauthClientId          string `required:"true" split_words:"true"`
-	OauthClientSecret      string `required:"true" split_words:"true"`
-	TlsCertificatePath     string `required:"true" split_words:"true"`
-	TlsPrivateKeyPath      string `required:"true" split_words:"true"`
-	TrustedUserEmailDomain string `required:"true" split_words:"true"`
-	SentryDsn              string `required:"false" split_words:"true"`
+	Port                   int    `toml:"port"`
+	InsecurePort           int    `toml:"insecure_port"`
+	DatabaseUrl            string `toml:"database_url"`
+	DataPath               string `toml:"data_path"`
+	Environment            string `toml:"environment"`
+	SharedSecret           string `toml:"shared_secret"`
+	OauthRedirectUrl       string `toml:"oauth_redirect_url"`
+	OauthClientId          string `toml:"oauth_client_id"`
+	OauthClientSecret      string `toml:"oauth_client_secret"`
+	TlsCertificatePath     string `toml:"tls_certificate_path"`
+	TlsPrivateKeyPath      string `toml:"tls_private_key_path"`
+	TrustedUserEmailDomain string `toml:"trusted_user_email_domain"`
+	SentryDsn              string `toml:"sentry_dsn"`
```

Comment:
I've converted this to snake_case, but you can see the noise it results in here, in the form of struct tags.